extensions/wikia/PowerUser/PowerUser.class.php

```diff
@@ -122,10 +140,10 @@ public function addPowerUserSetOption( $sProperty ) {
 	public function addPowerUserAddGroup( $sProperty ) {
 		if ( in_array( $sProperty, self::$aPowerUserProperties )
 			&& $this->bUseGroups
-			&& !in_array( self::GROUP_NAME, \UserRights::getGlobalGroups( $this->oUser ) )
+			&& !in_array( self::GROUP_NAME, $this->userPermissions()->getExplicitGlobalUserGroups( $this->oUser->getId() ) )
```

Comment:
I made the suggested change of unifying parameters - now all methods get the user objects instead of user id/user object. As for the proxy you suggested, I considered that, however part of methods take the user object for which we're performing a check, part take a user on which the operation is being performed and some both, so It wouldn't be possibly to do straightforward binding of the whole API to a single user object as its interpretation depends on the function context.